### PR TITLE
Add Merchant Sign In page

### PR DESCRIPTION
**Pull request by karenfrilya97**
Part of #2 

Changes made:
- Refactored out components of Sign Up that will be reused in Sign In.
- Added Sign In page using those components.

Differences from Merchant Sign Up page:
- No specific error message shown below each field, since there are only 2 fields.
- No checks for email format and password length, only if any field is empty.

image
The image above shows a screenshot of the Sign In page when a user has just filled in the email. The 'SIGN IN' button is disabled because there is an empty field.

### CODE REVIEW

File: web/src/components/merchant/sign-in/SignInCard.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * This is the card containing the sign in form for the Merchant app.
+ * It is displayed in the Sign In page.
+ */
+const SignInCard: React.FC = () => {
+  return (
+    <CardContainer>
+      <GroupBuyMerchantHeader />
+      <SignInForm />
+      <StyledRow>
+        Already have an account? <Link to="/merchant/sign-up">Sign up</Link>{' '}
```
Comment (hzxcaryn) [inline web/src/components/merchant/sign-in/SignInCard.tsx:53]: Do you think if we should use `&nbsp;`?

Comment (chshda) [inline web/src/components/merchant/sign-in/SignInCard.tsx:53]: The string should be not having an account?

Comment (karenfrilya97) [reply web/src/components/merchant/sign-in/SignInCard.tsx:53]: Oops, I forgot to change that! Thanks for pointing it out.

**Comment by karenfrilya97** [comment]
> SignInCard, SignInForm have very similar styles as SignUpCard and SignUpForm. The PageContainer in `merchant/sign-in/index.tsx` as well.
> Maybe can extract them out as common components?

Yeah! Good idea. I should do that.

File: web/src/components/merchant/sign-in/SignInForm.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        label="Email"
+        inputType="email"
+        forwardedRef={validations.email}
+      />
+      <FormRow
+        label="Password"
+        inputType="password"
+        forwardedRef={validations.password}
+      />
+      <ErrorContainer></ErrorContainer>
```
Comment (ifhuang) [inline web/src/components/merchant/sign-in/SignInForm.tsx:71]: no error msg?

File: web/src/components/merchant/sign-in/SignInCard.tsx (hunk 1 above)
Comment (xlai20) [reply web/src/components/merchant/sign-in/SignInCard.tsx:53]: +1. Can also add a reference back to sign in page from sign up page.

File: web/src/components/merchant/sign-in/SignInForm.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+import React from 'react';
+
+import FormRow from 'components/common/FormRow';
+import useSignInForm from 'components/merchant/sign-in/hooks/useSignInForm';
+import Button from 'muicss/lib/react/button';
+import Form from 'muicss/lib/react/form';
+import styled from 'styled-components';
+
+const StyledForm = styled(Form)`
```
Comment (xlai20) [inline web/src/components/merchant/sign-in/SignInForm.tsx:25]: Can you re-use the styles between sign-in and sign-up forms? They look so similar; some styles look identical (e.g. StyledForm).

File: web/src/components/merchant/sign-in/SignInCard.tsx (hunk 1 above)
Comment (karenfrilya97) [reply web/src/components/merchant/sign-in/SignInCard.tsx:53]: There already is a reference from sign up page to sign in page. It's a similar link below the SIGN UP button.

File: web/src/components/merchant/sign-in/SignInForm.tsx (hunk 2 above)
Comment (karenfrilya97) [reply web/src/components/merchant/sign-in/SignInForm.tsx:71]: I haven't added that yet. From the current checks, the only errors are caused by empty email or password. But in that case, I will just disable the SIGN IN button. Will add error after I add the actual signing up process with firebase auth.

File: web/src/components/merchant/sign-in/SignInForm.tsx (hunk 3 above)
Comment (karenfrilya97) [reply web/src/components/merchant/sign-in/SignInForm.tsx:25]: Okay! Will refactor similar components out to reduce duplication.

File: web/src/components/merchant/sign-in/SignInCard.tsx (hunk 1 above)
Comment (karenfrilya97) [reply web/src/components/merchant/sign-in/SignInCard.tsx:53]: Is it for the `{' '}`? But `&nbsp;` means non-breaking space right, and I don't need it to be non-breaking.
Maybe I can just remove the `{' '}`?

File: web/src/components/merchant/sign-in/SignInForm.tsx (hunk 2 above)
Comment (ifhuang) [reply web/src/components/merchant/sign-in/SignInForm.tsx:71]: I'm ok with disable button for empty email or password.
But I thought this error msg is for server call (submit) failure?

Comment (karenfrilya97) [reply web/src/components/merchant/sign-in/SignInForm.tsx]: Oh yepp, correct. Cos this PR hasn't done any server call yet.

**Pull request merged**